src/quartz_api/internal/inputs/dataplatform/client.py

```diff
@@ -0,0 +1,277 @@
+"""A data platform implementation that conforms to the DatabaseInterface."""
+
+from typing import override
+
+from dp_sdk.ocf import dp
+from fastapi import HTTPException
+
+from quartz_api import internal
+from quartz_api.internal.models import ForecastHorizon
+
+from ..utils import get_window
+
+
+class Client(internal.DatabaseInterface):
+    """Defines a data platform interface that conforms to the DatabaseInterface."""
+
+    c: dp.DataPlatformDataServiceStub
+
+    @classmethod
+    def new(cls, dp_client: dp.DataPlatformDataServiceStub) -> "Client":
```

Comment:
should this be renamed to `make_client` or something like that? `new` feels a bit funny